src/react/projects/spark-core-react/src/SprkCard/components/SprkCardTeaser/SprkCardTeaser.js

```diff
@@ -0,0 +1,267 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import SprkIcon from '../../../SprkIcon/SprkIcon';
+import SprkLink from '../../../SprkLink/SprkLink';
+import SprkButton from '../../../SprkButton/SprkButton';
+
+const SprkCardTeaser = props => {
+  const {
+    teaserConfig: {
+      additionalCtaClasses,
+      additionalCtaIconClasses,
+      additionalMediaIconClasses,
+      bodyText,
+      ctaAnalytics,
+      ctaIcon,
+      ctaText,
+      ctaType,
+      imgAlt,
+      imgSrc,
+      href,
+      linkType,
+      mediaIconName,
+      mediaLinkAnalytics,
+      mediaType,
+      title,
+      titleFirst,
+    },
+  } = props;
+
+  const BodyText = () => (
+    <p className="sprk-b-TypeBodyTwo sprk-o-Stack__item">{bodyText}</p>
+  );
+
+  const Title = () => (
+    <h3
+      className={cx('sprk-b-TypeDisplayFive sprk-o-Stack__item', {
+        'sprk-c-Card__content': titleFirst,
+      })}
+    >
+      {title}
+    </h3>
+  );
+
+  const Media = () => {
```

Comment:
I dont think we want to nest the functions like this. If we want to use the components outside of the context where they're defined, such as a new variant type, we'll want a flat structure.

Reply:
does this mean break these out into their own files?

Reply:
this was referring to how I was making a react component in side a react component (inside the larger react component) -- which obviously sounds wrong. This fixed itself during the data and link element refactor